server/controllers/entities/lib/resolver/update_resolved_entry.js

```diff
@@ -49,3 +59,39 @@ const addImageClaim = async (entity, imageUrl, newClaims) => {
   const { url: imageHash } = await getImageByUrl(imageUrl)
   newClaims['invp:P2'] = [ imageHash ]
 }
+
+const updateDatePrecision = (entity, updatedEntity, seedClaims) => {
+  const seedDateClaims = _.pick(seedClaims, simpleDayProperties)
+  Object.keys(seedDateClaims).forEach(prop => {
+    const seedDates = seedDateClaims[prop]
+    const seedDate = seedDates[0]
+    const currentDate = entity.claims[prop][0]
+    if (seedDate && currentDate && isMorePreciseDate(seedDate, currentDate) && doDatesAgree(seedDate, currentDate)) {
+      updatedEntity.claims[prop] = seedDateClaims[prop]
+    }
+  })
+}
+
+const simpleDayProperties = [ 'wdt:P569', 'wdt:P570', 'wdt:P571', 'wdt:P576', 'wdt:P577' ]
+
+const doDatesAgree = (date1, date2) => {
+  const date1Parts = dateParts(date1)
+  const date2Parts = dateParts(date2)
+  let mostPreciseDate, leastPreciseDate
+  if (date1Parts.length === date2Parts.length) return areDatesPartsEqual(date1Parts, date1Parts)
+  else if (date1Parts.length > date2Parts.length) {
+    mostPreciseDate = date1Parts
+    leastPreciseDate = date2Parts
+  } else {
+    mostPreciseDate = date2Parts
+    leastPreciseDate = date1Parts
+  }
+  const truncatedPreciseDate = mostPreciseDate.slice(0, leastPreciseDate.length)
+  return areDatesPartsEqual(truncatedPreciseDate, leastPreciseDate)
+}
```

Comment:
the simple-day format, in which month and days are 0-padded, allows to implement `doDatesAgree` as a simple string comparison:
```suggestion
const doDatesAgree = (seedDate, currentDate) => seedDate.startsWith(currentDate)
```

Reply:
ahah indeed ! i wanted to have an more agnostic comparison to compare whatever length on both side, but i guess YAGNI